fanficfare/adapters/adapter_fanfictionnet.py

```diff
@@ -63,14 +58,13 @@ def getAcceptDomains(cls):
     def getSiteExampleURLs(cls):
         return "https://www.fanfiction.net/s/1234/1/ https://www.fanfiction.net/s/1234/12/ http://www.fanfiction.net/s/1234/1/Story_Title http://m.fanfiction.net/s/1234/1/"
 
-    def set_story_idurl(self,url):
+    def set_story_id_and_url(self,url):
         parsedUrl = urlparse(url)
         pathparts = parsedUrl.path.split('/',)
         self.story.setMetadata('storyId',pathparts[2])
-        self.urltitle='' if len(pathparts)<5 else pathparts[4]
-        # normalized story URL.
-        self._setURL("https://"+self.getSiteDomain()\
-                         +"/s/"+self.story.getMetadata('storyId')+"/1/"+self.urltitle)
+        # normalized story URL
```

Comment:
Thanks for all the detailed responses. FYI, I appreciate you taking the stance of helping me to learn about the Python 2 vs 3 issues and the details of FanFicFare's architecture and coding style instead of being bothered by all the inappropriate code I throw at you.

It looks like this particular PR has too many problems to bother merging, and like you said, there is no hurry about getting the cache code into the main release. A nice thing about git is that this code will remain available for me to reference, so the work isn't for naught. I think I'll have some responses to your specific code comments.